sdk/options.go

```diff
@@ -21,13 +20,31 @@ type Options struct {
 
 // Limit sets the 'limit' Query parameter to the request
 func (o *Options) Limit(val string) *Options {
-	o.Query.Add(api.ParamLimit, val)
+	if o.Query == nil {
+		var queryParams url.Values = make(map[string][]string)
+		o.Query = queryParams
+	}
+	_, paramExists := o.Query[api.ParamLimit]
+	if paramExists {
+		o.Query.Set(api.ParamLimit, val)
+	} else {
+		o.Query.Add(api.ParamLimit, val)
+	}
```

Comment:
`Set()` will add if it doesn't exist anyway so you can just use Set without checking for existence. [See here for example](https://goplay.tools/snippet/HvcWaABZkTL).
```suggestion
	o.Query.Set(api.ParamLimit, val)
```

Reply:
that's good to know - done